src/Controls/src/Core/BindableObject.cs

```diff
@@ -31,7 +31,7 @@ public BindableObject()
 
 		readonly Dictionary<BindableProperty, BindablePropertyContext> _properties = new Dictionary<BindableProperty, BindablePropertyContext>(4);
 		bool _applying;
-		object _inheritedContext;
+		WeakReference _inheritedContext;
 
```

Comment:
Lastly, I found that `VisualElement.BackgroundProperty.BindingContext` held an instance of a `ViewModel`. This was the value of `Brush.Default._inheritedContext`.